<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="play_bag" default="true" />
  <arg name="bag_file" default="/home/victor/Desktop/hfloor_long2.bag"/>

  <arg name="mav_name" default="jay"/>
  <arg name="namespace" default="$(arg mav_name)" />

  <param name="use_sim_time" value="true" />

  <node name="player" pkg="rosbag" type="play" output="screen" args="-r 0.75 -s 0 --clock $(arg bag_file)" if="$(arg play_bag)">
    <!-- <remap from="/tf" to="/not_tf"/> -->
  </node>

  <group ns="$(arg namespace)" >
    <node name="rovio" pkg="rovio" type="rovio_node" output="screen">
      <param name="filter_config" value="$(find darpa_maplab_utils)/params/arche_25_01_2019/rovio_filter.info" />
      <param name="camera0_config" value="$(find darpa_maplab_utils)/params/arche_25_01_2019/rovio_cam0.yaml" />
      <param name="world_frame" value="odom"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="pose_lidar_broadcaster" args="-0.0109024 -0.0430394 -0.0872866 0.492577 0.492152 0.511611 -0.503397 imu os1 100">
      <param name="capability_group" value="Lidar" />
    </node>

    <node pkg="ouster_ros" type="motion_correction_node" name="ouster_motion_compensation" output="screen">
      <remap from="pointcloud" to="os1_node/points" />

      <param name="capability_group" value="Lidar" />
    </node>

    <node name="voxgraph_mapper" pkg="voxgraph" type="voxgraph_mapping_node" output="screen">
      <rosparam>
        verbose: false
        debug: true
        submap_creation_interval: 10
        pointcloud_topic: "ouster_motion_compensation/corrected_pointcloud"
        subscriber_queue_length: 1000
        world_frame: "odom"
        odom_tf_frame: "imu"
        T_odom_sensor:
        - [-0.0081478,  0.9999180,  0.0098774, -0.011]
        - [-0.0299520, -0.0101173,  0.9995002, -0.043]
        - [ 0.9995182,  0.0078479,  0.0300320, -0.087]
        - [ 0.0,        0.0,        0.0,        1.0]
        tsdf_integrator:
          truncation_distance: 0.20
          tsdf_voxel_size: 0.05
          max_ray_length_m: 10.0
          use_const_weight: true
      </rosparam>
    </node>
  </group>

  <!-- To generate the combined_mesh, please call: -->
  <!-- rosservice call /jay/voxgraph_mapper/publish_combined_mesh "{}" -->
  <arg name="rviz_file" default="$(find voxgraph)/cfg/voxgraph_mapper.rviz" />
  <node pkg="rviz" type="rviz" name="$(anon rviz)" respawn="false" args="-d $(arg rviz_file)" output="screen"/>
</launch>
